beard/clustering/blocking.py

```diff
@@ -88,6 +82,35 @@ def _parallel_fit(fit_, partial_fit_, estimator, verbose, data_queue,
     return
 
 
+def _single_fit(fit_, partial_fit_, estimator, verbose, data):
+    """Run clusterer's fit function."""
+    block, existing_clusterer = data
+    b, X, y = block
+
+    if len(X) == 1:
+        clusterer = _SingleClustering()
+    elif existing_clusterer and partial_fit_ and not fit_:
+        clusterer = existing_clusterer
+    else:
+        clusterer = clone(estimator)
+
+    if verbose > 1:
+        print("Clustering %d samples on block '%s'..." % (len(X), b))
+
+    if fit_ or not hasattr(clusterer, "partial_fit"):
+        try:
+            clusterer.fit(X, y=y)
+        except TypeError:
+            clusterer.fit(X)
+    elif partial_fit_:
+        try:
+            clusterer.partial_fit(X, y=y)
+        except TypeError:
+            clusterer.partial_fit(X)
+
+    return (b, clusterer)
```

Comment:
Obviously, some code could be factored out of `_single_fit` and `_parallel_fit`, but that is a minor issue.